Objects/abstract.c

```diff
@@ -819,6 +819,21 @@ binary_op(PyObject *v, PyObject *w, const int op_slot, const char *op_name)
     PyObject *result = binary_op1(v, w, op_slot);
     if (result == Py_NotImplemented) {
         Py_DECREF(result);
+
+        if (strcmp(op_name, ">>") == 0 && \
+                strcmp(v->ob_type->tp_name, "builtin_function_or_method") == 0) {
+            PyErr_Format(PyExc_TypeError,
+                 "unsupported operand type(s) for %.100s: "
+                 "'%.100s' and '%.100s'. Did you mean \"print(%s, "
+                 "file={:100!r}).format(%s))\"",
```

Comment:
I'm leaning towards the "hardcoded placeholders" option, as it occurs to me that we don't actually have access to the message to be displayed when raising the exception:

    print >> sys.stderr, 1, 2, 3

The LHS passed to `>>` is `print`, and the RHS is `sys.stderr`. I think that's why I just had `<expr>` as a placeholder in my example on the issue tracker.

However, the repr of a standard stream isn't particular helpful in this context either: 

    >>> sys.stderr
    <_io.TextIOWrapper name='<stderr>' mode='w' encoding='UTF-8'>

The generic `<output_stream>` placeholder would be clearer than displaying that.

So yeah, forget about interpolating the inputs at all, and just append a string with generic placeholders in it.